2018---The REAL First Power Up/src/org/usfirst/frc/team2586/robot/Robot.java

```diff
@@ -430,9 +435,9 @@ private boolean autoDrive(double distance) {
 		// so use the larger of the two (absolute distance)
 		double d;
 		if (Math.abs(l) > Math.abs(r))
-			d = l;
+			d = Math.abs(l);
 		else
-			d = r;
+			d = Math.abs(r);
 
```

Comment:
This looks wrong to me... 

If you overshoot your target, you need to apply a reverse power to get back to your target. This code will likely cause it to shoot off if it passes the target.

I'm guessing that the robot went the wrong way on the first attempt? Make kP negative to change the direction that the robot drives.

Reply:
Oh, this would also prevent you from driving backwards in auto.

Reply:
It seems to work correctly, for whatever reason the raw value was causing the initial 90 degree turn to work perfectly, but then it only turned back roughly 45 degrees. I assumed it was because having a negative value for the distance was messing with the turn function? I'll take a closer look

Reply:
I think I have it correct, I was having the problem that when I used auton drive one encoder was spitting a positive value and one encoder was spitting a negative value. If the negative value was used the conditions would never be met and it would keep driving indefinitely(if I remember correctly)

Reply:
Now that you mention it though, that might mean that the robot doesn't have the capability to drive backwards in autonomous... I haven't tested it

Reply:
Whichever encoder is counting backwards, change the scale factor to be negative. This way both will move positive when going forward.